kahuna/public/js/leases/leases.js

```diff
@@ -87,31 +94,56 @@ leases.controller('LeasesCtrl', [
                     : "This image already has a syndication lease. It will be overwritten. " +
                       "Do you wish to proceed?";
                   const shouldApplyLeases = $window.confirm(confirmText);
-                  if (!shouldApplyLeases) {
+                    if (!shouldApplyLeases) {
+                      //We're not saving it if the user cancels, so clear up
+                    //   return ctrl.updateLeases();
+                    ctrl.adding = false;
                     return;
                   }
                 }
 
+
+
               leaseService.batchAdd(ctrl.newLease, ctrl.images)
                 .catch((e) => {
                   console.error(e);
                   alertFailed('Something went wrong when saving, please try again.');
                 })
-                .finally(() => {
+                  .finally(() => {
+                      console.log("HELO");
                     ctrl.resetLeaseForm();
                 });
             }
         };
 
         // These events allow this control to work as a hybrid on the upload page
+        // Update from future: not very well.
```

Comment:
Is this any better now?

Reply:
It's still a massively dangerous pattern.

Reply:
To rephrase 😁 does a developer in the future need this comment? If so, will they understand it as it stands?

Reply:
Reworded.